cohort-cli/src/main/java/com/ibm/cohort/cli/input/Parameter.java

```diff
@@ -7,26 +7,56 @@
 
 import static org.apache.commons.lang.StringUtils.isEmpty;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.ibm.cohort.annotations.Generated;
 
 public class Parameter {
-	@JsonProperty("name")
-	private String name;
+	private static final String INTERVAL_TYPE = "interval";
 
-	@JsonProperty("type")
+	private String name;
 	private String type;
-
-	@JsonProperty("value")
 	private String value;
-
-	@JsonProperty("subtype")
 	private String subtype;
-
-	@JsonProperty("start")
 	private String start;
-
-	@JsonProperty("end")
 	private String end;
+	
+	/**
+	 * 
+	 * @param name Name of parameter in the CQL
+	 * @param type CQL type of the parameter
+	 * @param value CQL-formatted string value of the parameter
+	 */
+	public static Parameter create(String name, String type, String value) {
+		Parameter p = new Parameter(name, type, value, null, null, null);
+		p.validate();
+		return p;
+	}
+
+	/**
+	 * 
+	 * @param name Name of parameter in the CQL
+	 * @param subtype Type of interval for the parameter
+	 * @param start String representation of the start of the interval
+	 * @param end String representation of the end of the interval
+	 */
+	public static Parameter createInterval(String name, String subtype, String start, String end) {
+		Parameter p = new Parameter(name, INTERVAL_TYPE, null, subtype, start, end);
+		p.validate();
+		return p;
+	}
```

Comment:
My original design in cohort-engine for parameter handling was that applications consuming the API would pass in the strongly-typed objects from the CQL engine JARs. The string-based stuff was intended to handle CLI invocations where the only interface was a String. I think we drifted away from that with the implementation of the JSON interface. Is there a value in making the API users jump through loosely typed values to get to strongly typed parameter data?

Reply:
> Is there a value in making the API users jump through loosely typed values to get to strongly typed parameter data?
The only advantage I can think of right now is that `ParameterHelper` at least yells at you if you pick illegal types for your parameters (and then creates strongly typed objects in a way  we expect).

If we want to steer folks right to strongly  typed objects do we want to offer any sort of code to make that  process easier for users? Or is it up to them to know the cql parameter types and make sure they are building their strongly typed objects properly?

Reply:
I guess I could just point at `ParameterHelper` as a reference, but steer folks to making the parameter objects on their own.

Reply:
The biggest disadvantage I see with the strongly typed objects is that they carry implementation details about the underlying CQL engine implementation along with them and that it would be harder to swap out the engine impl for a different one at some unspecified later point. It is hard to say how likely that is.

Do we know what format the IHE application is going to have the parameter data in at the point where they are calling us? If they already have them as strings, maybe this isn't all that important to address right now. I certainly wouldn't block this PR.